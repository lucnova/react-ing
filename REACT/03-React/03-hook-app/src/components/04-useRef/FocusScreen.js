import React, { useRef } from 'react';

export const FocusScreen = () => {
    // * USO COMÃšN ES PARA GUARDAR REFERENCIA A ALGO.
    //      En este caso solo se usa para tener referencia del elemento input
    //      aunque es mas preferible hacer un querySelector
	const inputRef = useRef();

	const handleFocusOnInput = () => {
        inputRef.current.select();

        console.log(inputRef);

        //document.querySelector('input').select();
    };

	return (
		<>
			<div className="row">
				<div className="col pt-5 px-5">
					<h1>FocusScreen</h1>

					<hr />
				</div>
			</div>

			<div className="row">
				<div className="col pt-5 px-5">
					<input ref={inputRef} className="form-control" placeholder="Nombre"></input>

					<button className="btn btn-primary mt-3" onClick={handleFocusOnInput}>
						OK
					</button>
				</div>
			</div>
		</>
	);
};
