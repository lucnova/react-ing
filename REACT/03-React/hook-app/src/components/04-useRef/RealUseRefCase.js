import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-useFetch/MultipleCustomHooks';

export const RealUseRefCase = () => {
	const [showQuotes, setShowQuotes] = useState(false);

	const handleQuoteToggle = () => {
		setShowQuotes(!showQuotes);
	};

	return (
		<>
			<div className="row">
				<div className="col pt-5 px-5">
					<h1>RealUseRefCase</h1>

					<hr />
				</div>
			</div>

			<div className="row">
				<div className="col mb-2 px-5">
					<button className="btn btn-success" onClick={handleQuoteToggle}>
						Mostrar / Ocultar
					</button>
				</div>
			</div>

			{showQuotes && (
				<div className="row">
					<div className="col">
						<MultipleCustomHooks></MultipleCustomHooks>
					</div>
				</div>
			)}
		</>
	);
};
