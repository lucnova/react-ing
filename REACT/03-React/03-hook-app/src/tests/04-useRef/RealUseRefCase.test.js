import { shallow } from 'enzyme';
import { RealUseRefCase } from '../../components/04-useRef/RealUseRefCase';

describe('Testing on RealUseRefCase', () => {
	const wrapper = shallow(<RealUseRefCase />);

	test('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
		// * NO debe existir MultipleCustomHooks
		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
	});

	test('should display MultipleCustomHooks component when button is clicked', () => {
		// * Simular un click
		wrapper.find('button').simulate('click');
		// * Debe existir MultipleCustomHooks
		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
	});
});
