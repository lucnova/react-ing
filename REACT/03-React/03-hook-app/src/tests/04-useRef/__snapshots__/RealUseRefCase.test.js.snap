// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on RealUseRefCase should match snapshot 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col pt-5 px-5"
    >
      <h1>
        RealUseRefCase
      </h1>
      <hr />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col mb-2 px-5"
    >
      <button
        className="btn btn-success"
        onClick={[Function]}
      >
        Mostrar / Ocultar
      </button>
    </div>
  </div>
</Fragment>
`;
